tests/check_check_sub.c

```diff
@@ -27,6 +27,17 @@
 #include <check.h>
 #include "check_check.h"
 
+#if !defined(NAN)
```

Comment:
As mentioned earlier, is there a way to put this in libcompat.h?

Reply:
The definition can be moved there.
But the definition alone is not enough - there needs to be some code to initialize the value. You need to generate the NAN and INFINITY at runtime, basically at start time of the project. Currently there is no initialization function in libcompat.c.
I can try if this can be done with some preprocessor magic, or by nesting some initializations.
But it's probably not really working if there is no initialization function.